### PR TITLE
JBPM-5233: Data source management

### PR DESCRIPTION
**Pull request by wmedvede**
- adds configuration parameters for Widlfy10 domain mode
    - adds a change handler component for ensuring deployment updates when running un cluster mode

### CODE REVIEW

**Comment by wmedvede** [comment]
Hello @manstis, would you mind review this PR?
Thanks,
Walter.

File: kie-wb-common-screens/kie-wb-common-datasource-mgmt/kie-wb-common-datasource-mgmt-backend/src/main/java/org/kie/workbench/common/screens/datasource/management/backend/service/handler/AbstractDefChangeHandler.java [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+            } else {
+                def = DriverDefSerializer.deserialize( ioService.readAllString( Paths.convert( path ) ) );
+            }
+        } catch ( Exception e ) {
+            logger.error( "It was not possible to deserialize content from file: " + path, e );
+        }
+        return def;
+    }
+
+    private boolean isDriverFile( Path path ) {
```
Comment (manstis) [inline kie-wb-common-screens/kie-wb-common-datasource-mgmt/kie-wb-common-datasource-mgmt-backend/src/main/java/org/kie/workbench/common/screens/datasource/management/backend/service/handler/AbstractDefChangeHandler.java]: Duplicated in [DefResourceChangeObserver](https://github.com/droolsjbpm/kie-wb-common/blob/46a326a935d6982b19574b6ee4cfc93541dfcd5e/kie-wb-common-screens/kie-wb-common-datasource-mgmt/kie-wb-common-datasource-mgmt-backend/src/main/java/org/kie/workbench/common/screens/datasource/management/backend/service/DefResourceChangeObserver.java#L72)

File: kie-wb-common-screens/kie-wb-common-datasource-mgmt/kie-wb-common-datasource-mgmt-backend/src/main/java/org/kie/workbench/common/screens/datasource/management/backend/service/handler/AbstractDefChangeHandler.java [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+        return path.getFileName( ).endsWith( ".driver" );
+    }
+
+    private boolean isDataSourceFile( Path path ) {
```
Comment (manstis) [inline kie-wb-common-screens/kie-wb-common-datasource-mgmt/kie-wb-common-datasource-mgmt-backend/src/main/java/org/kie/workbench/common/screens/datasource/management/backend/service/handler/AbstractDefChangeHandler.java]: Duplicated in [DefResourceChangeObserver](https://github.com/droolsjbpm/kie-wb-common/blob/46a326a935d6982b19574b6ee4cfc93541dfcd5e/kie-wb-common-screens/kie-wb-common-datasource-mgmt/kie-wb-common-datasource-mgmt-backend/src/main/java/org/kie/workbench/common/screens/datasource/management/backend/service/DefResourceChangeObserver.java#L76)

File: kie-wb-common-screens/kie-wb-common-datasource-mgmt/kie-wb-common-datasource-mgmt-backend/src/main/java/org/kie/workbench/common/screens/datasource/management/backend/service/DefResourceChangeObserver.java [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    private DefChangeHandler defChangeHandler;
+
+    public DefResourceChangeObserver( ) {
+    }
+
+    public void setDefChangeHandler( DefChangeHandler defChangeHandler ) {
+        this.defChangeHandler = defChangeHandler;
+    }
+
+    public void onResourceAdd( @Observes final ResourceAddedEvent resourceAddedEvent ) {
```
Comment (manstis) [inline kie-wb-common-screens/kie-wb-common-datasource-mgmt/kie-wb-common-datasource-mgmt-backend/src/main/java/org/kie/workbench/common/screens/datasource/management/backend/service/DefResourceChangeObserver.java]: OK, I see how it's setup [here](https://github.com/droolsjbpm/kie-wb-common/pull/577/files#diff-40699caafdbbd55df8a156c3247f8987R96).

**Comment by wmedvede** [comment]
Hello @manstis, regarding your query about using the CDI events, the answer is yes, this is something we agreed with @csadilek to use.

File: kie-wb-common-screens/kie-wb-common-datasource-mgmt/kie-wb-common-datasource-mgmt-backend/src/main/java/org/kie/workbench/common/screens/datasource/management/backend/service/DefResourceChangeObserver.java [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+                    resourceRenamedEvent.getDestinationPath( ), resourceRenamedEvent.getSessionInfo( ) );
+        }
+    }
+
+    public void onResourceDelete( @Observes final ResourceDeletedEvent resourceDeletedEvent ) {
+        if ( defChangeHandler != null && isProcessable( resourceDeletedEvent.getPath( ) ) ) {
+            defChangeHandler.processResourceDelete( resourceDeletedEvent.getPath( ), resourceDeletedEvent.getSessionInfo( ) );
+        }
+    }
+
```
Comment (wmedvede) [inline kie-wb-common-screens/kie-wb-common-datasource-mgmt/kie-wb-common-datasource-mgmt-backend/src/main/java/org/kie/workbench/common/screens/datasource/management/backend/service/DefResourceChangeObserver.java]: @manstis At this moment think there's no need for processing batch events. Even if the datasource definition is a file, it also represents "live" thing that is managed by the wb, etc., and we don't expect people manipulating the datsources manually, batch modifications, etc.

**Comment by wmedvede** [comment]
Hello @manstis the suggestions were applied. I've rebased with master to be sure it compiles against the latest commits. From my side it's ready to merge.

Thanks.

**Pull request merged**